File: library-manager-ui/src/services/systemSettingService.js
import { axiosPrivate } from '~/apis/configHttp';

export const updateLibraryRules = (values) => {
    return axiosPrivate.put('system-settings/library-rules', values);
};

export const getLibraryRules = () => {
    return axiosPrivate.get('system-settings/library-rules');
};

export const getAllHolidays = (params) => {
    return axiosPrivate.get(`system-settings/holidays?${params}`);
};

export const getHolidayById = (id) => {
    return axiosPrivate.get(`system-settings/holidays/${id}`);
};

export const addHoliday = (values) => {
    return axiosPrivate.post('system-settings/holidays', values);
};

export const updateHoliday = (id, values) => {
    return axiosPrivate.put(`system-settings/holidays/${id}`, values);
};

export const deleteHoliday = (id) => {
    return axiosPrivate.delete(`system-settings/holidays/${id}`);
};

export const getLibraryConfig = () => {
    return axiosPrivate.get('system-settings/library-config');
};

export const updateLibraryConfig = (values) => {
    return axiosPrivate.put('system-settings/library-config', values);
};

File: library-manager-server/src/main/java/com/example/librarymanager/service/SystemSettingService.java
package com.example.librarymanager.service;

import com.example.librarymanager.domain.dto.common.CommonResponseDto;
import com.example.librarymanager.domain.dto.request.HolidayRequestDto;
import com.example.librarymanager.domain.dto.request.LibraryConfigRequestDto;
import com.example.librarymanager.domain.dto.request.LibraryRulesRequestDto;
import com.example.librarymanager.domain.dto.response.HolidayResponseDto;
import com.example.librarymanager.domain.dto.response.LibraryConfigResponseDto;

import java.util.List;

public interface SystemSettingService {
    CommonResponseDto updateLibraryRules(LibraryRulesRequestDto requestDto, String userId);

    String getLibraryRules();

    List<HolidayResponseDto> getAllHolidays(Boolean activeFlag);

    HolidayResponseDto getHolidayById(String id);

    CommonResponseDto addHoliday(HolidayRequestDto holidayRequestDto, String userId);

    CommonResponseDto updateHoliday(String id, HolidayRequestDto holidayRequestDto, String userId);

    CommonResponseDto deleteHoliday(String id);

    LibraryConfigResponseDto getLibraryConfig();

    CommonResponseDto updateLibraryConfig(LibraryConfigRequestDto requestDto, String userId);
}
